e2e-tests/single-machine/single_machine_deployment_validation.sh

```diff
@@ -0,0 +1,314 @@
+function usage() {
+  echo "This script validates if number of resources captured through"
+  echo "parquet files match what is stored in the source FHIR server"
+  echo
+  echo " usage: ./single_machine_deployment_validation.sh  HOME_DIR  PARQUET_SUBDIR  [OPTIONS] "
+  echo "    HOME_DIR          Path where e2e-tests/dingle-machine directory is. Directory MUST"
+  echo "                      contain the parquet tools jar as well as subdirectory"
+  echo "                      of parquet file output"
+  echo "    PARQUET_SUBDIR    Subdirectory name under HOME_DIR containing"
+  echo "                      parquet files  "
+  echo
+  echo " Options:  "
+  echo "     --use_docker_network     Flag to specify whether to use docker"
+  echo "                              or host network URLs"
+}
+
+#################################################
+# Makes sure args passed are correct
+#################################################
+function validate_args() {
+  if [[ $# -lt 2 || $# -gt 5  ]]; then
+    echo "Invalid number of args passed."
+    usage
+    exit 1
+  fi
+
+  echo "Checking if the Parquet-tools JAR exists..."
+  if compgen -G "${1}/parquet-tools*.jar" > /dev/null; then
+    echo "Parquet-tools JAR exists in ${1}"
+  else
+    echo "Parquet-tools JAR not found in ${1}"
+    usage
+    exit 1
+  fi
+}
+
+#################################################
+# Function that prints messages
+# Arguments:
+#   anything that needs printing
+#################################################
+function print_message() {
+  local print_prefix="E2E TEST FOR SINGLE MACHINE DEPLOYMENT:"
+  echo "${print_prefix} $*"
+}
+
+#################################################
+# Function that defines the global vars
+# Globals:
+#   HOME_PATH
+#   PARQUET_SUBDIR
+#   SOURCE_FHIR_SERVER_URL
+#   PIPELINE_CONTROLLER_URL
+#   THRIFTSERVER_URL
+# Arguments:
+#   Path where e2e-tests/single-machine directory is. Directory contains parquet tools jar as
+#      well as subdirectory of parquet file output
+#   Subdirectory name under HOME_DIR containing parquet files.
+#      Example: dwh
+#   Optional: Flag to specify whether to use docker or host network URLs.
+#################################################
+function setup() {
+  HOME_PATH=$1
+  PARQUET_SUBDIR=$2
+  SOURCE_FHIR_SERVER_URL='http://localhost:8091'
+  PIPELINE_CONTROLLER_URL='http://localhost:8090'
+  THRIFTSERVER_URL='localhost:10001'
+  if [[ $3 = "--use_docker_network" ]]; then
+    SOURCE_FHIR_SERVER_URL='http://hapi-server:8080'
+    PIPELINE_CONTROLLER_URL='http://pipeline-controller:8080'
+    THRIFTSERVER_URL='thriftserver:10000'
+  fi
+}
+
+#################################################
+# Function to count resources in source fhir server
+# Globals:
+#   HOME_PATH
+#   PARQUET_SUBDIR
+#   SOURCE_FHIR_SERVER_URL
+#   TOTAL_TEST_PATIENTS
+#   TOTAL_TEST_ENCOUNTERS
+#   TOTAL_TEST_OBS
+#################################################
+function fhir_source_query() {
```

Comment:
I have refactored a little bit like removing openmrs and streaming related things. And I think if we create a common script file which will have just this function (because this is the only common function), including it will be a challenge. Like if we have created lets say common.sh in single-machine folder then we will have to give relative path for it to include in pipeline_validation.sh and copy the same in Dockerfile as well. Or, we maintain two copies for this which increases complexities.

Reply:
resolving this for now